src/main/java/org/reactome/addlinks/fileprocessors/HmdbMetabolitesFileProcessor.java

```diff
@@ -130,4 +129,54 @@ public HmdbMetabolitesFileProcessor()
 		return null;
 	}
 
+	
+	private void transformXmlToTsv(String inputFilename, String outfileName)
+	{
+		
+		try(OutputStream outStream = new FileOutputStream(outfileName);)
+		{
+			XMLInputFactory xif = XMLInputFactory.newInstance();
+			XMLStreamReader xsr = xif.createXMLStreamReader(new FileReader(inputFilename));
+			xsr.nextTag();
+			Source source = new StreamSource(this.getClass().getClassLoader().getResourceAsStream("hmdb_metabolites_transform.xsl"));
+			TransformerFactory factory = TransformerFactory.newInstance();
+			Transformer transformer = factory.newTransformer(source);
+			
+			while (xsr.nextTag() == XMLStreamConstants.START_ELEMENT)
+			{
+				if (xsr.getName().getLocalPart().equals(METABOLITE_ELEMENT_NAME))
+				{
```

Comment:
It was pretty intuitive to me what was going on in this method until this part. Could you include a small description of what specifically happens once the 'metabolite' tag is found?

Reply:
Comments were added.